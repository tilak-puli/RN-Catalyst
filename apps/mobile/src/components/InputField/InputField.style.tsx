import {TextInput} from 'react-native';
import styled from 'styled-components/native';

export const StyledText = styled(TextInput)`
  height: 43px;
  font-size: 14px;
  border-radius: 5px;
  border-width: 1px;
  border-color: #eaeaea;
  background-color: #fafafa;
  padding-left: 10px;
  margin: 5px 0;
`;
