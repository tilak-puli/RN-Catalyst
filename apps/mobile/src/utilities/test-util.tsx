import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {SpinnerProvider} from '../context/spinner/spinnerContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <QueryClientProvider client={query}>
    <SpinnerProvider>{children}</SpinnerProvider>
  </QueryClientProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
