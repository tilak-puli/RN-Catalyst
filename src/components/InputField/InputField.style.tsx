import {makeStyles} from '@rneui/themed';

const useStyles = makeStyles(theme => ({
  text: {
    height: 43,
    fontSize: 14,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: theme.colors.white1,
    backgroundColor: theme.colors.white0,
    paddingLeft: 10,
    marginVertical: 5,
  },
}));

export default useStyles;
