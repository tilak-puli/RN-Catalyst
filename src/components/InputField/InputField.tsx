import {TextInput, TextInputProps, View} from 'react-native';
import useStyles from './InputField.style';

const InputField = ({style, ...rest}: TextInputProps) => {
  const styles = useStyles();
  return (
    <View>
      <TextInput
        style={[styles.text, style]}
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...rest}
      />
    </View>
  );
};

export default InputField;
