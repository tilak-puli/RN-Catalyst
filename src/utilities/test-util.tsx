import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '@rneui/themed';
import {SpinnerProvider} from '../context/spinner/spinnerContext';
import {GlobalContextProvider} from '../context/global/globalContext';
import theme from '../styles/theme';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <GlobalContextProvider>
    <ThemeProvider theme={theme}>
      <QueryClientProvider client={query}>
        <SpinnerProvider>{children}</SpinnerProvider>
      </QueryClientProvider>
    </ThemeProvider>
  </GlobalContextProvider>
);

export * from '@testing-library/react-native';
export {customRender as render};
