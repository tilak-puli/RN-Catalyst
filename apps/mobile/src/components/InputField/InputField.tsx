import {TextInputProps, View} from 'react-native';
import {StyledText} from './InputField.style';

const InputField = ({style, ...rest}: TextInputProps) => (
  <View>
    <StyledText
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...rest}
      style={style}
    />
  </View>
);

export default InputField;
