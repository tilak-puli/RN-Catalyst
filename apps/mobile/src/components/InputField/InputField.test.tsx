import InputField from './InputField';
import {render, screen} from '../../utilities/test-util';
import 'jest-styled-components/native';

describe('[Component] - [InputField] ', () => {
  test('should render text input correctly', () => {
    const input = render(
      <InputField placeholder="Username" accessibilityLabel="Username" />,
    );
    expect(input).toMatchSnapshot();
  });

  test('should match background color', () => {
    render(
      <InputField
        testID="test-input"
        placeholder="Username"
        accessibilityLabel="Username"
      />,
    );

    expect(screen.getByTestId('test-input')).toHaveStyleRule(
      'background-color',
      '#fafafa',
    );
  });
});
