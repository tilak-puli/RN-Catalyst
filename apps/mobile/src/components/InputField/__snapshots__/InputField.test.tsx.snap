// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - [InputField]  should render text input correctly 1`] = `
<View>
  <TextInput
    accessibilityLabel="Username"
    placeholder="Username"
    style={
      Array [
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
          "paddingLeft": 10,
        },
      ]
    }
  />
</View>
`;
